src/runtime/components/Link.vue

```diff
@@ -130,7 +132,23 @@ const ui = computed(() => tv({
   }, appConfig.ui?.link || {})
 }))
 
-const to = computed(() => props.to ?? props.href)
+const to = computed(() => {
+  const path = props.to ?? props.href
+  if (!path) return path
+
+  // Skip external links and absolute URLs
+  if (props.external || (typeof path === 'string' && hasProtocol(path, { acceptRelative: true }))) {
+    return path
+  }
+
+  // Use localePath from `@nuxtjs/i18n` if available
+  const localePath = nuxtApp.$localePath as ((route: RouteLocationRaw) => RouteLocationRaw) | undefined
+  if (localePath) {
+    return localePath(path)
+  }
```

Comment:
```suggestion
  const localeRoute = nuxtApp.$localeRoute as ((route: RouteLocationRaw) => RouteLocationRaw) | undefined
  if (localeRoute) {
    return localeRoute(path)
  }
```

`localePath` will resolve to a path `string`, using `localeRoute` should preserve route properties (things like `state`)

Reply:
Thanks @BobbieGoede 😊